test/y2firstboot/clients/licenses_test.rb

```diff
@@ -44,53 +45,71 @@
 
     allow(Yast::WFM).to receive(:CallFunction)
       .and_return(client_response)
+
+    allow(Yast::GetInstArgs).to receive(:argmap)
+      .and_return({"directory" => directory})
   end
 
   describe "#run" do
-    context "when FIRSTBOOT_LICENSE_DIR is defined" do
-      let(:firstboot_license_dir) { "/path/to/licenses" }
-      let(:firstboot_novell_license_dir) { "" }
+    context "when 'directory' argument is given" do
+      let(:directory) { "/path/to/somewhere" }
       let(:expected_arg) do
-        { "directories" => [firstboot_license_dir] }
+        { "directories" => [directory] }
       end
 
-      it "includes it as arg for InstLicense client" do
+      it "includes it as an argument for InstLicense client" do
         expect(Yast::WFM).to receive(:CallFunction)
-          .with("inst_license", array_including(hash_including(expected_arg)))
+          .with("inst_license", array_including(hash_including("directory" => directory)))
```

Comment:
NP: why not `expected_arg`?

Reply:
Because `expected_arg` is a leftover since I removed it just to be more explicit.

Reply:
Just above I see this:

~~~
      let(:expected_arg) do
        { "directories" => [directory] }
      end
~~~

So we could use it if we adapt the code to always use `directories` key.

Reply:
> Just above I see this:
> ...

Yes, because I forgot to remove it. That's why said that it is intentionally to use the explicit expectation and the `expected_arg` is a leftover.